frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -1044,6 +1045,13 @@
                             vm.showSubmissionMetaAttributesOnLeaderboard = true;
                         }
 
+                        for (var k = 0; k < vm.leaderboard[i].leaderboard__schema.labels.length; k++) {
```

Comment:
Might wanna use this

var leaderboardLabels = vm.leaderboard[i].leaderboard__schema.labels;
var index = leaderboardLabels.findIndex(label => label === vm.orderLeaderboardBy);
var choosenmatrix = index !== -1 ? [index.toString()] : ['0'];